### PR TITLE
Date Range Error

### LINKED ISSUE #151: Date Range component
## Bug Report

**Current behavior:**
- **Incorrect range selection**

If you have an end-date selected (screenshot - `2018-02-14`), and clear the start-date, you will be able to select any date in the calendar as the start-date, even the dates after `2018-02-14`.

screen shot 2017-12-27 at 8 51 31 pm

Also, if I have an end date selected, and I'm selecting the start date, the calendar should grey-out irrelevant dates (i.e. the dates after the end date)

- **UX issues**

    * Currently, the component does not tell you in any way which date are you selecting at any given point. (i.e. you won't know if you're selecting an end-date or a start-date)
    * The range selection isn't continuous. Calendar closes after selecting the start-date and you will have to click the end-date input field to open the calendar again to select the end date.

- **Support more Customization**

Allow users to change the middle `-`'s markup which connects two input fields together. Possible usecase: You may want to change the `-` to an arrow (`->`) or `to`.

**Expected / new behavior:**

- To improve the UX, we can highlight the range while selecting and keep the calendar open until the user is done selecting both the dates. 

**Reactivesearch version:**  latest

### PR DESCRIPTION
**Pull request by jyash97**
For #151

### CODE REVIEW

**Comment by divyanshu013** [comment]
Hi @jyash97, thanks for contributing 🤝

The UX issue is fixed with selecting date beyound the end date.

From the issue, here are the things which are missing before I can close this:

> Currently, the component does not tell you in any way which date are you selecting at any given point. (i.e. you won't know if you're selecting an end-date or a start-date)

For this you could implement the hover effect. [Example from docs](http://react-day-picker.js.org/examples/selected-range-enter)

> Allow users to change the middle -'s markup which connects two input fields together. Possible usecase: You may want to change the - to an arrow (->) or to.

> For custom rendering, we can use `onDayChange` from react-day-picker to expose the current selection and render as we please with a custom trigger to open the calendar view.

> When an end date is selected and the start date is cleared, we should still show the end date as marked partially.

Currently, it displays like the following:

image

You could implement a similar behavior which we have when only the start date is selected like:

image

(Here the starting date is partially selected to give a visual cue)

**Comment by jyash97** [comment]
Highlighting the `End Date` :

screen shot 2018-06-27 at 11 10 53 am

And for the hovering effect, I think we will need to Change the methods, to onMouseEnter and onDayClick instead of onDayChange.

**Comment by divyanshu013** [comment]
Yes, check out the example I shared (http://react-day-picker.js.org/examples/selected-range-enter) on how they're doing it. A similar approach should work here IMO, you might need some new variables in state to keep track of the hovered date range.

**Comment by jyash97** [comment]
I tried it but the function mouse enter is not working.

**Comment by divyanshu013** [comment]
Did you try this one? http://react-day-picker.js.org/api/DayPicker#onDayMouseEnter

**Comment by jyash97** [comment]
Yes I tried this one and I checked by putting a console message on the first line and it didn't executed the message.

**Comment by metagrover** [comment]
Can you try upgrading `react-day-picker` to the latest version and see if that helps?

**Comment by metagrover** [comment]
Can you try upgrading `react-day-picker` to the latest version and see if that helps?

Also, we need `onDayMouseEnter` on `DayPicker` (not `DayPickerInput`) for hover to work.

**Comment by jyash97** [comment]
In the component `DayPickerInput ` is used so may I change it to `DayPicker`?

**Comment by divyanshu013** [comment]
You could use `dayPickerProps` to pass props to the `DayPicker` rendered by `DayPickerInput` http://react-day-picker.js.org/api/DayPickerInput#dayPickerProps

**Comment by divyanshu013** [comment]
@jyash97 the hover effect isn't working, rest of the things look good. Also there are linting errors in the file.

File: packages/web/src/components/date/DateRange.js [hunk 1]
```diff
@@ -212,6 +213,12 @@ class DateRange extends Component {
 		});
 	}
 
+	handleDayMouseEnter = (day) => {
+		this.setState({
+			enteredTo: day,
```
Comment (divyanshu013) [inline packages/web/src/components/date/DateRange.js]: This variable isn't being used, did you mean to use `dateHovered`?

Comment (jyash97) [reply packages/web/src/components/date/DateRange.js]: Yes sorry, I forgot to change this name, it should be `dateHovered` and everything would then be working

Comment (divyanshu013) [reply packages/web/src/components/date/DateRange.js]: I tested but it wasn't could you test it out first?

**Comment by jyash97** [comment]
screen1

**Comment by divyanshu013** [comment]
Thanks for contributing @jyash97 ✌️

**Pull request merged**